druid/src/widget/widget_ext.rs

```diff
@@ -93,6 +95,43 @@ pub trait WidgetExt<T: Data>: Widget<T> + Sized + 'static {
         SizedBox::new(self).expand()
     }
 
+    /// Wrap this widget in a [`ConstrainedBox`]. The function provided can
+    /// return arbitrary BoxConstraints, so please constrain responsibly.
+    ///
+    /// [`ConstrainedBox`]: struct.ConstrainedBox.html
+    fn constrain(
+        self,
+        constrain: impl Fn(&BoxConstraints) -> BoxConstraints + 'static,
+    ) -> ConstrainedBox<T> {
+        ConstrainedBox::new(self, Box::new(constrain))
+    }
+
+    /// Wrap this widget in a [`ConstrainedBox`]. This mutates the widget's
+    /// minimum BoxConstraints height.
+    ///
+    /// [`ConstrainedBox`]: struct.ConstrainedBox.html
+    fn min_height(self, height: f64) -> ConstrainedBox<T> {
+        self.constrain(move |bc: &BoxConstraints| {
+            BoxConstraints::new(
+                (bc.min().width, height).into(),
+                (bc.max().width, (bc.max().height).max(height)).into(),
+            )
+        })
+    }
+
+    /// Wrap this widget in a [`ConstrainedBox`]. This mutates the widget's
+    /// minimum BoxConstraints width.
+    ///
+    /// [`ConstrainedBox`]: struct.ConstrainedBox.html
+    fn min_width(self, width: f64) -> ConstrainedBox<T> {
```

Comment:
these make me feel like we should *some* validation. It would be very easy to pass these some value that is larger than max width, and I think we should at least log that?

Reply:
I'm actually guarding against that with: `(bc.max().height).max(height)` but I could log a warning instead? (bc.debug_check also complains about it)

Reply:
that sounds good. Maybe there should be a method on `BoxConstraints` that constraints _other_ box constraints? 🤔

Reply:
Yeah that makes sense to me.